src/overlays/actors/ovl_En_Light/z_en_light.c

```diff
@@ -21,15 +27,181 @@ const ActorInit En_Light_InitVars = {
+        Lights_InitType2PositionalLight(&this->posLightInfo, this->actor.posRot.pos.x,
+                                        yOffset + (s16)this->actor.posRot.pos.y, this->actor.posRot.pos.z, 255, 255,
+                                        180, -1);
+    }
+
+    this->lightNode = Lights_Insert(globalCtx, &globalCtx->lightCtx, &this->posLightInfo);
+    Actor_SetScale(&this->actor, D_80A9E840[this->actor.params & 0xF].scale * 0.0001f);
+    this->timer = (s32)(Math_Rand_ZeroOne() * 255.0f);
+
+    if ((this->actor.params & 0x400) != 0) {
+        this->actor.update = EnLight_UpdateSwitch;
+    }
+}
+
+void EnLight_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnLight* this = THIS;
+
+    Lights_Remove(globalCtx, &globalCtx->lightCtx, this->lightNode);
+}
+
+void EnLight_UpdatePosRot(EnLight* this, GlobalContext* globalCtx) {
+    // update yaw for billboard effect
+    this->actor.shape.rot.y = func_8005A9F4(ACTIVE_CAM) + 0x8000;
+
+    if (this->actor.attachedA != NULL) {
+        Math_Vec3f_Copy(&this->actor.posRot.pos, &(this->actor.attachedA)->posRot.pos);
+        this->actor.posRot.pos.y += 17.0f;
+    }
+
+    this->timer++;
+}
+
+void EnLight_Update(Actor* thisx, GlobalContext* globalCtx) {
+    f32 intensity;
+    FlameParams* flameParams;
+    s16 radius;
+    EnLight* this = THIS;
+
+    flameParams = &D_80A9E840[this->actor.params & 0xF];
+    intensity = (Math_Rand_ZeroOne() * 0.5f) + 0.5f;
+    radius = (this->actor.params < 0) ? 100 : 300;
+    Lights_SetPositionalLightColorAndRadius(&this->posLightInfo, (flameParams->primColor.r * intensity),
+                                            (flameParams->primColor.g * intensity),
+                                            (flameParams->primColor.b * intensity), radius);
+    EnLight_UpdatePosRot(this, globalCtx);
+
+    if (this->actor.params >= 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_TORCH - SFX_FLAG);
+    }
+}
+
+void EnLight_UpdateSwitch(Actor* thisx, GlobalContext* globalCtx) {
+    f32 intensity;
+    FlameParams* flameParams;
+    EnLight* this = THIS;
+    f32 scale;
+
+    flameParams = &D_80A9E840[this->actor.params & 0xF];
+    scale = this->actor.scale.x / ((f32)flameParams->scale * 0.0001);
+
+    if (this->actor.params & 0x800) {
+        if (Flags_GetSwitch(globalCtx, (this->actor.params & 0x3F0) >> 4)) {
+            Math_ApproxF(&scale, 1.0f, 0.05f);
+        } else {
+            if (scale < 0.1f) {
+                Actor_SetScale(&this->actor, 0.0f);
+                return;
+            }
+            Math_ApproxF(&scale, 0.0f, 0.05f);
+        }
+    } else {
+        if (Flags_GetSwitch(globalCtx, (this->actor.params & 0x3F0) >> 4)) {
+            if (scale < 0.1f) {
+                Actor_SetScale(&this->actor, 0.0f);
+                return;
+            }
+            Math_ApproxF(&scale, 0.0f, 0.05f);
+        } else {
+            Math_ApproxF(&scale, 1.0f, 0.05f);
+        }
+    }
```

Comment:
```suggestion
    } else if (Flags_GetSwitch(globalCtx, (this->actor.params & 0x3F0) >> 4)) {
        if (scale < 0.1f) {
            Actor_SetScale(&this->actor, 0.0f);
            return;
        }
        Math_ApproxF(&scale, 0.0f, 0.05f);
    } else {
        Math_ApproxF(&scale, 1.0f, 0.05f);
    }
```

Reply:
`else if` seems worse since it breaks the balance with the equivalent code in the first `if`

Reply:
i normally agree with these, but in this case the else is pertaining to the opposite case with params, while the flag check is nested inside like the case above

Reply:
idk if i worded that properly but
if (params check)
    if (flag check)
else
    if (flag check)

makes more sense to me and is more "symmetrical" i guess than making the flag check an else if of the params check.. if that makes sense.

If others dont agree tho i can change it

Reply:
Oh, didn't realise this was something we preferred. Never mind this then.

Reply:
its a case by case basis i dont think we really have set rules on it. i just think it makes more sense in this scenario even tho i understand why it can seem messy